helm/crossplane/templates/provider-install/controller-config-aws.yaml

```diff
@@ -0,0 +1,11 @@
+{{ if eq .Values.giantswarm.kubernetes.provider.kind "aws" }}
+apiVersion: pkg.crossplane.io/v1alpha1
+kind: ControllerConfig
+metadata:
+  name: "{{- include "provider.aws.name" . }}"
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  {{- with .Values.giantswarm.crossplane.providers.aws.controllerConfig }}
+    {{- . | toYaml | nindent 2 }}
```

Comment:
My security concern is basically these lines in the controller configs. There are a lot of possible values: https://doc.crds.dev/github.com/crossplane/crossplane/pkg.crossplane.io/ControllerConfig/v1alpha1@v1.9.1, for example setting a service account or security context. And this basically renders everything you pass as value, which makes is flexible and much easier, cleaner to implement.

I tried to mitigate this in the `values.yaml` by setting only a few possible values and then in the `values.schema.yaml` I set no additional properties allowed. So in theory chart operator will reject if someone want to set not ok / supported fields. Another small problem with that is  when you run `helm schema-gen` it will overwrite the whole file and potentially removes these fields which might be easy to miss in a PR review.